### ISSUE TITLE
Installing TelFit on M1 Chip Mac

### ISSUE THREAD

**Issue by ericasaw** [kgullikson88/Telluric-Fitter#38]
I have been having a few issues with installing TelFit both through the GitHub master branch and through the pip install. Both currently result in the same error from line 235 in setup.py. I should also note that I am trying to install TelFit on one of the new M1 chip Macs, but I have been able to successfully install all of the dependencies with the correct version number so far. Also just wanted to mention that I completely destroyed any install of TelFit from GitHub before I tried using pip install and vice versa. Lastly, I did try trying to sudo pip install since I thought it might be a permissions error somewhere, but I get the same error code.

ERROR: Command errored out with exit status 1:
   command: /opt/miniconda3/envs/igrins_rv/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/6x/3qkmq8256ss373jsgcxm5csw0000gn/T/pip-install-aggpt5mb/telfit_28defd87e7174963bed0aeea2afb70f8/setup.py'"'"'; __file__='"'"'/private/var/folders/6x/3qkmq8256ss373jsgcxm5csw0000gn/T/pip-install-aggpt5mb/telfit_28defd87e7174963bed0aeea2afb70f8/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/6x/3qkmq8256ss373jsgcxm5csw0000gn/T/pip-wheel-trlu43i_
       cwd: /private/var/folders/6x/3qkmq8256ss373jsgcxm5csw0000gn/T/pip-install-aggpt5mb/telfit_28defd87e7174963bed0aeea2afb70f8/
  Complete output (46 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.9-x86_64-3.7
  copying src/TelluricFitter.py -> build/lib.macosx-10.9-x86_64-3.7
  copying src/MakeModel.py -> build/lib.macosx-10.9-x86_64-3.7
  copying src/DataStructures.py -> build/lib.macosx-10.9-x86_64-3.7
  copying src/MakeTape5.py -> build/lib.macosx-10.9-x86_64-3.7
  creating build/lib.macosx-10.9-x86_64-3.7/telfit
  copying src/telfit/__init__.py -> build/lib.macosx-10.9-x86_64-3.7/telfit
  running build_ext
  Un-packing aer_v_3.2.tar.gz
  aer_v_3.2/line_file/aer_v_3.2: truncated gzip input
  tar: Error exit delayed from previous errors.
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/private/var/folders/6x/3qkmq8256ss373jsgcxm5csw0000gn/T/pip-install-aggpt5mb/telfit_28defd87e7174963bed0aeea2afb70f8/setup.py", line 355, in <module>
      package_dir={'': 'src'}
    File "/opt/miniconda3/envs/igrins_rv/lib/python3.7/site-packages/setuptools/__init__.py", line 153, in setup
      return distutils.core.setup(**attrs)
    File "/opt/miniconda3/envs/igrins_rv/lib/python3.7/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/opt/miniconda3/envs/igrins_rv/lib/python3.7/distutils/dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "/opt/miniconda3/envs/igrins_rv/lib/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/opt/miniconda3/envs/igrins_rv/lib/python3.7/site-packages/wheel/bdist_wheel.py", line 299, in run
      self.run_command('build')
    File "/opt/miniconda3/envs/igrins_rv/lib/python3.7/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/opt/miniconda3/envs/igrins_rv/lib/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/opt/miniconda3/envs/igrins_rv/lib/python3.7/distutils/command/build.py", line 135, in run
      self.run_command(cmd_name)
    File "/opt/miniconda3/envs/igrins_rv/lib/python3.7/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/opt/miniconda3/envs/igrins_rv/lib/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/private/var/folders/6x/3qkmq8256ss373jsgcxm5csw0000gn/T/pip-install-aggpt5mb/telfit_28defd87e7174963bed0aeea2afb70f8/setup.py", line 291, in run
      MakeLBLRTM()
    File "/private/var/folders/6x/3qkmq8256ss373jsgcxm5csw0000gn/T/pip-install-aggpt5mb/telfit_28defd87e7174963bed0aeea2afb70f8/setup.py", line 235, in MakeLBLRTM
      subprocess.check_call(["tar", "-xzf", '{}{}'.format(TELLURICMODELING, fname), '-C', TELLURICMODELING])
    File "/opt/miniconda3/envs/igrins_rv/lib/python3.7/subprocess.py", line 363, in check_call
      raise CalledProcessError(retcode, cmd)
  subprocess.CalledProcessError: Command '['tar', '-xzf', '/Users/ericasaw/.TelFit/aer_v_3.2.tar.gz', '-C', '/Users/ericasaw/.TelFit/']' returned non-zero exit status 1.

When I tried to install from just downloading the code on GitHub, I was able to comment out that line in setup.py and I got a completely different error because the script wasn't able to find any of the makefiles to compile TelFit (I can provide this error message too, figured I would just start with the pip install version error first). Please let me know if I can provide any more information!

**Comment by ericasaw** [comment]
I think I figured out where the issue in pip installing comes from--line 68 in setup.py contains a stale link so all of the required files for install are not downloaded which leads to the failure on line 235 because the downloaded files do not exist. This could probably be easily fixed by replacing the link with one that will work for downloading the files (I found the files at https://zenodo.org/record/1202479/#.YNnvShNucq1).

After a few hours of playing with this I was able to successfully install TelFit from the master branch (v.1.4.0), here is the step by step of what I did if anyone runs into the same issues:

1. Comment out line 235 in setup.py--as previously mentioned this link on line 68 is stale so the download fails which triggers the error from this line, we manually fix this later.
2. Run `python setup.py build` in the TelFit directory and wait for failure
3. Download the tar files from https://zenodo.org/record/1202479/#.YNnvShNucq1 and move them into the ~/.TelFit directory
4. Run 'python setup.py build' again and it should compile.

If for whatever reason you want to try to install v.1.3.1 this is how you fix the install:

1. Comment out line 213 in setup.py--as previously mentioned the download link is stale so the download fails which triggers the error from this line, we manually fix this later.
2.  Run `python setup.py build` in the TelFit directory and wait for failure
3. Download the tar files from https://zenodo.org/record/1202479/#.YNnvShNucq1 and move them into the ~/.TelFit directory
4. Change line 210 in setup.py from `for fname in ['aer_v_3.2.tar.gz', 'aerlnfl_v2.6.tar.gz', 'aerlbl_v12.2.tar.gz']:` to `for fname in ['aer_v_3.2.tar.gz', 'aerlnfl_v2.6.tar.gz', 'aerlbl_v12.2.1.tar.gz']:`
5. Run `python setup.py build` in the TelFit directory and wait for failure
6. Edit the makefile.common in the ~/.TelFit/lnfl/build and ~/.TelFit/lblrtm/build directories to include the -std=legacy flag in the FCFLAG line for your corresponding compiler (mine for using GNU Fortran 11.1.0 on the Mac M1 chip was osxGNUsgl)
7. manually compile the makefile in the corresponding directories using `make -f make_lblrtm osxGNUsgl` or `make -f make_lnfl osxGNUsgl`
8. In setupy.py change line 343 from `ext_modules=[Extension("FittingUtilities", ["src/FittingUtilities.c"],` to `ext_modules=[Extension("FittingUtilities", ["src/FittingUtilities.pyx"],`
9. Run `python setup.py build` in the TelFit directory and the build should work!

Also note, if you cannot get version 1.4.0 to install using the fixes above, but you can get 1.3.1 to install, you can get to version 1.4.0 by deleting your master directory github download and sudo rm -r the ~/.TelFit directory and then downloading the master 1.4.0 version from github and running python setup.py build. Even though version 1.4.0 has download issues (the link doesn't work when trying to download the required files--line 68) for some reason installing 1.3.1 first and removing the directories and reinstalling seems to build without issue as well (haven't really looked into this much, just recently found this--probably has something to do with not completely cleaning the prior install).

Hope that helps!

**Comment by gully** [comment]
Hi @ericasaw thank you for reporting your solution for successfully installing TelFit on an M1 Mac 🙏   I was able to verify that the link on Line 68 in `setup.py` is now working when combined with the filenames on line 231.  These lines were changed right around the same time as your comment so hopefully most folks attempting to install TelFit on M1 Macs will not have to experience the same friction.  Thanks again for reporting this Issue.
